src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -37,6 +35,38 @@ public abstract partial class JsonTypeInfo
         private Action<object>? _onDeserializing;
         private Action<object>? _onDeserialized;
 
+        internal JsonTypeInfo(Type type, JsonConverter converter, JsonSerializerOptions options)
+        {
+            Type = type;
+            Options = options;
+            Converter = converter;
+            PropertyInfoForTypeInfo = CreatePropertyInfoForTypeInfo();
+            ElementType = converter.ElementType;
+
+            switch (converter.ConverterStrategy)
+            {
+                case ConverterStrategy.Dictionary:
+                    {
+                        KeyType = converter.KeyType;
```

Comment:
could we just set it always similarly to ElementType and then get rid of switch statement which seems can be replaced with `if (converter.ConverterStrategy == ConverterStrategy.None)`?

Reply:
Looks like it. This switch statement used to be doing much more work in the past but could be removed now and replaced with the `GetTypeInfoKind` method.